library/cc/stream.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+#include <vector>
+
+#include "library/common/types/c_types.h"
+#include "request_headers.h"
+#include "request_trailers.h"
+
+class Stream {
+public:
+  Stream(envoy_stream_t handle);
+
+  Stream& send_headers(RequestHeadersSharedPtr headers, bool end_stream);
+  Stream& send_data(const std::vector<uint8_t>& data);
```

Comment:
After thinking about this a bit, I think we should switch to using `envoy_data` directly in this implementation to represent buffers. Python's buffer protocol doesn't work with vectors, nor do any buffer representations in Envoy. We've already have a bridge implementation of Envoy's internal buffer type defined in terms of `envoy_data`. It seems like using vectors will just result in extra indirection, and potentially, copies.